packages/core/src/routes/connector.ts

```diff
@@ -122,11 +122,28 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
         })
       );
 
+      const insertConnectorId = generateConnectorId();
       ctx.body = await insertConnector({
-        id: generateConnectorId(),
+        id: insertConnectorId,
         ...body,
       });
 
+      if (
+        connectorFactory.type === ConnectorType.Sms ||
+        connectorFactory.type === ConnectorType.Email
+      ) {
+        const logtoConnectors = await getLogtoConnectors();
+        const allSameTypeConnectorsIds = logtoConnectors
+          .filter((logtoConnector) => logtoConnector.type === connectorFactory.type)
+          .map((logtoConnector) => logtoConnector.dbEntry.id)
```

Comment:
Looks like this mapping is redundant.

Reply:
We need to get the ids of all conflicting connectors.